pkg/parser/validate/workflow_test.go

```diff
@@ -94,3 +94,143 @@ workflows:
 
 	CheckYamlErrors(t, testCases)
 }
+
+func TestWorkflowMaxAutoReruns(t *testing.T) {
+	testCases := []ValidateTestCase{
+		{
+			Name: "Valid max_auto_reruns value 1",
+			YamlContent: `version: 2.1
+
+workflows:
+  test-workflow:
+    max_auto_reruns: 1
+    jobs:
+      - hold:
+          type: approval`,
+			OnlyErrors:  true,
+			Diagnostics: []protocol.Diagnostic{},
+		},
+		{
+			Name: "Valid max_auto_reruns value 3",
+			YamlContent: `version: 2.1
+
+workflows:
+  test-workflow:
+    max_auto_reruns: 3
+    jobs:
+      - hold:
+          type: approval`,
+		},
+		{
+			Name: "Valid max_auto_reruns value 5",
+			YamlContent: `version: 2.1
+
+jobs:
+	test-job:
+		docker:
+			- image: cimg/base:stable
+		steps:
+			- run: echo "test"
+
+workflows:
+	test-workflow:
+		max_auto_reruns: 5
+		jobs:
+			- test-job`,
+			OnlyErrors:  true,
+			Diagnostics: []protocol.Diagnostic{},
+		},
+		{
+			Name: "Invalid max_auto_reruns value 0 (below minimum)",
+			YamlContent: `version: 2.1
+
+jobs:
+  test-job:
+    docker:
+      - image: cimg/base:stable
+    steps:
+      - run: echo "test"
+
+workflows:
+  test-workflow:
+    max_auto_reruns: 0
+    jobs:
+      - test-job`,
+			OnlyErrors: true,
+			Diagnostics: []protocol.Diagnostic{
+				{
+					Range: protocol.Range{
+						Start: protocol.Position{Line: 11, Character: 21},
+						End:   protocol.Position{Line: 11, Character: 22},
+					},
+					Severity: protocol.DiagnosticSeverityError,
+					Source:   "cci-language-server",
+					Message:  "Must be greater than or equal to 1",
+					Data:     []protocol.CodeAction{},
+				},
+			},
+		},
+		{
+			Name: "Invalid max_auto_reruns value 6 (above maximum)",
+			YamlContent: `version: 2.1
+
+jobs:
+  test-job:
+    docker:
+      - image: cimg/base:stable
+    steps:
+      - run: echo "test"
+
+workflows:
+  test-workflow:
+    max_auto_reruns: 6
+    jobs:
+      - test-job`,
+			OnlyErrors: true,
+			Diagnostics: []protocol.Diagnostic{
+				{
+					Range: protocol.Range{
+						Start: protocol.Position{Line: 11, Character: 21},
+						End:   protocol.Position{Line: 11, Character: 22},
+					},
+					Severity: protocol.DiagnosticSeverityError,
+					Source:   "cci-language-server",
+					Message:  "Must be less than or equal to 5",
+					Data:     []protocol.CodeAction{},
+				},
+			},
+		},
+		{
+			Name: "Invalid max_auto_reruns string value",
+			YamlContent: `version: 2.1
+
+jobs:
+  test-job:
+    docker:
+      - image: cimg/base:stable
+    steps:
+      - run: echo "test"
+
+workflows:
+  test-workflow:
+    max_auto_reruns: "invalid"
+    jobs:
+      - test-job`,
+			OnlyErrors: true,
+			Diagnostics: []protocol.Diagnostic{
+				{
+					Range: protocol.Range{
+						Start: protocol.Position{Line: 11, Character: 21},
+						End:   protocol.Position{Line: 11, Character: 30},
+					},
+					Severity: protocol.DiagnosticSeverityError,
+					Source:   "cci-language-server",
+					Message:  "Must be greater than or equal to 1",
```

Comment:
~ is this error message produced by the json schema validator? I would think that the error would be the value is not an integer which would be a better error message.

Reply:
No it's an error from https://github.com/CircleCI-Public/circleci-yaml-language-server/pull/348/files#diff-125a976b310086ded94dbe71b972da679e3b5051603aacdb9ac9af710144f365R20. Iit seems like the schema doesn't affect validation - if I remove the validation code change, the max/mins on the schema don't seem to do anything (the tests that expect errors fail)
The schema is used for hovering according to https://github.com/SchemaStore/schemastore/blob/master/src/schemas/json/circleciconfig.json, but I guess not validation?
I can ask lifecycle if any of them know.